### PR TITLE
Update documentation on the revert on env validation

### PR DESCRIPTION
**Pull request by vdemeester**
Follow-up to #16608 where I did forget to update the documentation :sweat:.

/cc @moxiegirl @thaJeztah @calavera 

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

### CODE REVIEW

File: docs/reference/commandline/run.md [hunk 1]
```diff
@@ -243,9 +247,9 @@ An example of a file passed with `--env-file`
     helloWorld=true
     TEST_PASSTHROUGH=
     HOME=/root
+    123qwe=bar
+    org.spring.config=something
 
-> **Note**: Environment variables names must consist solely of letters, numbers,
```
Comment (vdemeester) [inline docs/reference/commandline/run.md]: I'm wondering if I remove that line *only" or if I replace it to something like :


> Environment variable names used by the utilities in the Shell and Utilities volume of IEEE Std 1003.1-2001 consist solely of uppercase letters, digits, and the '_' (underscore) from the characters defined in Portable Character Set and do not begin with a digit. **Other characters may be permitted by an implementation; applications shall tolerate the presence of such names**. Uppercase and lowercase letters shall retain their unique identities and shall not be folded together. The name space of environment variable names containing lowercase letters is reserved for applications.

File: docs/reference/commandline/run.md [hunk 2]
```diff
@@ -216,7 +216,9 @@ An example of a file passed with `--env-file`
     _TEST_BAR=FOO
     TEST_APP_42=magic
     helloWorld=true
-    # 123qwe=bar <- is not valid
+    # The two following are not valid in the  "posix" sence but will get passed
```
Comment (phemmer) [inline docs/reference/commandline/run.md]: The posix comment is technically incorrect. Posix is quite happy to allow those variable names. What it says is that the shell functions and utilities covered within posix must not use variable names such as those (eg, `sh` is defined in posix, and if you wish to be posix compliant, your implementation of `sh` cannot use a variable called `123qwe`).

Comment (vdemeester) [inline docs/reference/commandline/run.md]: @phemmer yeah you're right, I think the comment does not add anything useful anyway.. I'll remove it :stuck_out_tongue:

File: docs/reference/commandline/run.md (hunk 1 above)
Comment (moxiegirl) [inline docs/reference/commandline/run.md]: @vdemeester I think that might be overkill.  Maybe just a link to the standard?

Comment (vdemeester) [inline docs/reference/commandline/run.md]: @moxiegirl sounds good to me :wink:

**Comment by vdemeester** [comment]
Updated according to your comments (@moxiegirl @phemmer)

File: docs/reference/commandline/run.md [hunk 3]
```diff
@@ -243,9 +246,12 @@ An example of a file passed with `--env-file`
...
     TEST_PASSTHROUGH=
     HOME=/root
+    123qwe=bar
+    org.spring.config=something
 
-> **Note**: Environment variables names must consist solely of letters, numbers,
-> and underscores - and cannot start with a number.
+> **Note**: Environment variables are passed as-is, see
+> [here](http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap08.html)
+> for more information about it.
```
Comment (phemmer) [inline docs/reference/commandline/run.md]: I would vote for dropping the note entirely? Do we really need something saying "we don't mess with the data you gave us"?
Though that technically inaccurate, as we strip leading spaces.

Comment (vdemeester) [inline docs/reference/commandline/run.md]: True, updated :wink:

**Comment by vdemeester** [comment]
Updated :)

**Comment by thaJeztah** [comment]
LGTM, thanks!

**Comment by thaJeztah** [comment]
ping @moxiegirl @SvenDowideit

**Comment by moxiegirl** [comment]
LGTM Thanks @vdemeester !

**Pull request merged**